### PR TITLE
Change placement from enum to an integer

### PR DESCRIPTION
**Pull request by rkimballn1**

### CODE REVIEW

File: src/ngraph/graph_util.hpp [hunk 1]
```diff
@@ -420,7 +419,7 @@ namespace ngraph
 
     // Assert that nodes in the function is colocated and return that placement
```
Comment (diyessi) [inline src/ngraph/graph_util.hpp:420]: Might as well fix the comment while in the neighborhood.

File: src/ngraph/node.hpp [hunk 2]
```diff
@@ -428,10 +427,10 @@ namespace ngraph
         bool has_same_type(std::shared_ptr<const Node> node) const;
 
         /// Get device placement
```
Comment (diyessi) [inline src/ngraph/node.hpp:429]: Need a comment explaining now placement is used.

File: src/ngraph/node.hpp [hunk 3]
```diff
@@ -545,7 +544,7 @@ namespace ngraph
         std::deque<descriptor::Input> m_inputs;
         std::deque<descriptor::Output> m_outputs;
         std::unordered_map<Node*, autodiff::Adjoints> m_adjoint_map;
-        Placement m_placement = Placement::DEFAULT;
+        int32_t m_placement = -1;
```
Comment (diyessi) [inline src/ngraph/node.hpp:547]: Make a const for this magic -1 value.

File: src/ngraph/graph_util.hpp (hunk 1 above)
Comment (rkimballn1) [reply src/ngraph/graph_util.hpp:420]: done

**Pull request merged**